work/v1/types.go

```diff
@@ -66,6 +70,72 @@ type DeleteOption struct {
 	SelectivelyOrphan *SelectivelyOrphan `json:"selectivelyOrphans,omitempty"`
 }
 
+// ManifestConfigOption represents the configurations of a manifest defined in workload field.
+type ManifestConfigOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// +kubebuilder:validation:Required
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
```

Comment:
remember that in a v2 API, it would be a better choice to combine this with a `Manifest` itself to have spacial locality. We cannot add it now because `Manifest` is an inline rawExtension.  Sins of the past.

Reply:
yes, I will consider this change to work-api in sigs